File: BOJ/02XXX/2751-수_정렬하기(2).py
'''/**
* ABOUT
*
* DATE: 2021-05-12
* AUTHOR: lovelysi0113
*
* URL: https://www.acmicpc.net/problem/2751
*
* COMMENT:
* 수를 하나씩 받아서 배열에 넣은다음, sort해주면 된다(시간 복잡도: O(nlogn))
* 시간을 더 줄이기 위해 input() 대신 sys.stdin.readline()을 사용할 수 있다
**/'''

import sys

nums = int(input())

arr = []
for _ in range(0, nums):
    arr.append(int(sys.stdin.readline()))

arr.sort()

for el in arr:
    print(el)

File: BOJ/02XXX/2751-수_정렬하기(2).js
/**
* ABOUT
*
* DATE: 2021-05-12
* AUTHOR: lovelysi0113
*
* URL: https://www.acmicpc.net/problem/2751
*
* COMMENT:
* 수를 하나씩 받아서 배열에 넣은다음, sort해주면 된다(시간 복잡도: O(nlogn))
**/

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];

rl.on('line', function (line) {
    input.push(line);
})
.on('close', function () {
    let num = parseInt(input[0]);
    
    let arr = [];
    for (let idx = 1; idx <= num; idx++) {
        arr.push(parseInt(input[idx]));
    }
    
    arr.sort((a, b) => {
        if (a > b) return 1;
        else if (a === b) return 0;
        else return -1;
    });
    
    console.log(arr.join('\n'));
    
    process.exit();
});